engines/experiment/runner/runner.go

```diff
@@ -23,4 +32,8 @@ type ExperimentRunner interface {
 		payload []byte,
 		options GetTreatmentOptions,
 	) (*Treatment, error)
+	RegisterMetricsCollector(
```

Comment:
We need to update the dev docs for the RPC plugin ([here](https://github.com/caraml-dev/turing/blob/main/engines/experiment/docs/rpc_plugins.md)) with the details of the new method. This can be done in a separate PR too. But, in this one, it will be good to add some inline comments for the 2 runner methods to capture their summary, especially that the `RegisterMetricsCollector` method is only called on startup.

Reply:
Yep that makes sense, given all the changes we're making to the experiment engine interfaces. I've just added inline comments to the methods and structs introduced to this file. I'll add additional dev docs to capture more details in a separate PR.